jena-db/jena-tdb2/src/main/java/org/apache/jena/tdb2/xloader/ProcIngestDataX.java

```diff
@@ -103,9 +117,41 @@ public static void exec(String location,
         SystemIRIx.setProvider(provider);
     }
 
+    private static DatasetGraph getDatasetGraph(String location) {
+        Location loc = Location.create(location);
+        // Ensure reset
+        DatasetGraph dsg0 = DatabaseMgr.connectDatasetGraph(location);
+        TDBInternal.expel(dsg0);
+
+        StoreParams dftStoreParams = StoreParams.getDftStoreParams();
+        StoreParams storeParams = dftStoreParams;
+
+        if ( true ) {
+            storeParams = StoreParams.builder(StoreParams.getDftStoreParams())
+                    .node2NodeIdCacheSize(10_000_000)
+                    .build();
+        }
+
+        DatasetGraph dsg = DatabaseConnection.connectCreate(loc, storeParams).getDatasetGraph();
+        StoreParams storeParamsActual = TDBInternal.getDatasetGraphTDB(dsg).getStoreParams();
+
+        FmtLog.info(LOG, "Node to NodeId cache size: %,d", storeParamsActual.getNode2NodeIdCacheSize());
+        FmtLog.info(LOG, "NodeId to Node cache size: %,d", storeParamsActual.getNodeId2NodeCacheSize());
+
+        return dsg;
+    }
+
     private static Pair<Long, Long> build(DatasetGraph dsg, ProgressMonitor monitor,
                               OutputStream outputTriples, OutputStream outputQuads,
                               List<String> datafiles) {
+
+        int node2NodeIdCacheSize = StoreParamsConst.Node2NodeIdCacheSize; // 200k
+        node2NodeIdCacheSize = 1_000_000_000;// 1B!
+
+        StoreParams storeParmas =
+                StoreParams.builder().node2NodeIdCacheSize(node2NodeIdCacheSize)
+                .build();
```

Comment:
This set of `storeParams` doesn't obviously get used AFAICT

Reply:
Just after the "if" in `DatabaseConnection.connectCreate(loc, storeParams)`.

Reply:
Yes which is back at Line 135 using the `storeParams` created at Line 130

The `storeParams` created here at Line 151 aren't subsequently used (you're using the `dsg` passed in which itself uses the parameters from Line 135)

Should what's here at Line 151 not just be included in the parameters built at Line 135?

Reply:
Ah - OK - I'll tidy up in an other PR.  Several places can be tidied up.